classes/assessment/assessment.php

```diff
@@ -223,4 +223,12 @@ protected function get_equivalent_grade_from_mark(float $marks): ?string {
         }
         return $equivalentgrade;
     }
+
+    /**
+     * Get all participants for the assessment.
+     *
+     * @return array
+     */
+    abstract public function get_all_participants(): array;
```

Comment:
is this used by anything?

Reply:
It just set the rule, requires every assessment class extends this class must have this function.